deploy/docker-ephemeral/build/Makefile

```diff
@@ -0,0 +1,286 @@
+	sed -i.bak "s=\(409B6B1796C275462A1703113804BB82D39DC0E3\)=\1 7D2BAF1CF37B13E2069D6956105BD0E739499BDB=" $@
+
+airdock_fakesqs/Dockerfile:
+	git clone git@github.com:airdock-io/docker-fake-sqs.git airdock_fakesqs-all
+	cd airdock_fakesqs-all && git reset --hard $(AIRDOCKFAKESQS_COMMIT)
+	cp -R airdock_fakesqs-all/0.3.1 airdock_fakesqs
+	sed -i.bak "s=airdock/rvm:latest=$(USERNAME)/airdock_rvm$(TAGNAME)=" $@
+# add a workdir declaration to the final switch to root.
+	sed -i.bak "s=^USER root=USER root\nWORKDIR /=" $@
+# break directory creation into two pieces, one run by root.
+	sed -i.bak "s=^USER ruby=USER root=" $@
+	sed -i.bak "s=cd /srv/ruby/fake-sqs.*=chown ruby.ruby /srv/ruby/fake-sqs\nUSER ruby\nWORKDIR /srv/ruby/fake-sqs\nRUN cd /srv/ruby/fake-sqs \&\& \\\\=" $@
+
+java_maven_node_python/Dockerfile:
+	git clone git@github.com:localstack/localstack.git java_maven_node_python
+	cd java_maven_node_python && git reset --hard $(JAVAMAVENNODEPYTHON_COMMIT)
+	cd java_maven_node_python && mv bin/Dockerfile.base Dockerfile
+# disable installing docker-ce. not available on many architectures in binary form.
+	sed -i.bak "/.*install Docker.*/{N;N;N;N;N;d}" $@
+
+localstack/Dockerfile:
+	git clone git@github.com:localstack/localstack.git localstack
+	cd localstack && git reset --hard $(LOCALSTACK_COMMIT)
+	sed -i.bak "s=localstack/java-maven-node-python=$(USERNAME)/java_maven_node_python$(TAGNAME)=" $@
+# skip tests. they take too long.
+	sed -i.bak "s=make lint.*=make lint=" localstack/Makefile
+	sed -i.bak "s=\(.*lambda.*\)=#\1=" localstack/Makefile
+
+minio/Dockerfile:
+	git clone git@github.com:minio/minio.git minio
+	cd minio && git reset --hard $(MINIO_COMMIT)
+
+cassandra/Dockerfile:
+	git clone git@github.com:docker-library/cassandra.git cassandra-all
+	cd cassandra-all && git reset --hard $(CASSANDRA_COMMIT)
+	cp -R cassandra-all/3.11 cassandra
+	 # work around go compiler bug by using newer version of GOSU. https://bugs.launchpad.net/qemu/+bug/1696353
+	sed -i.bak "s/GOSU_VERSION .*/GOSU_VERSION 1.11/" $@
+	 # add a block to the entrypoint script to interpret CS_JVM_OPTIONS, modifying the jvm.options before launching cassandra.
+	 # first, add a marker to be replaced before the last if.
+	sed -i.bak -r ':a;$$!{N;ba};s/^(.*)(\n?)fi/\2\1REPLACEME\nfi/' cassandra/docker-entrypoint.sh
+	 # next, load our variables.
+	sed -i.bak 's/REPLACEME/\nAPP_CONFIG="$$CASSANDRA_CONFIG"\n&/' cassandra/docker-entrypoint.sh
+	sed -i.bak 's/REPLACEME/JVM_OPTIONS="$$CS_JVM_OPTIONS"\n&/' cassandra/docker-entrypoint.sh
+	 # add our parser and replacer.
+	sed -i.bak $$'s@REPLACEME@if [ ! -z "$${JVM_OPTIONS}" ]; then\\nfor x in $${JVM_OPTIONS}; do { l="$${x%%=*}"; r=""; e=""; [ "$$x" != "$${x/=//}" ] \&\& e="=" \&\& r="$${x##*=}"; [ "$$x" != "$${x##-Xm?}" ] \&\& r="$${x##-Xm?}" \&\& l="$${x%%$$r}"; echo $$l $$e $$r; _sed-in-place "$$APP_CONFIG/jvm.options" -r \'s/^[# ]*(\'"$$l$$e"\').*/\\\\1\'"$$r"\'/\'; } done\\nfi\\n&@' cassandra/docker-entrypoint.sh
```

Comment:
Is there any possible way we can simplify this? Can any human on earth read this? haha 😋

Reply:
This is all addressed in make-docker-and-qemu.md. all of the evil tricks in this makefile are explained in detail. :)